src/Signature/SignatureV4.php

```diff
@@ -14,21 +14,27 @@ class SignatureV4 implements SignatureInterface
 {
     use SignatureTrait;
     const ISO8601_BASIC = 'Ymd\THis\Z';
+    const UNSIGNED_PAYLOAD = 'UNSIGNED-PAYLOAD';
 
     /** @var string */
     private $service;
 
     /** @var string */
     private $region;
 
+    /** @var bool */
+    private $unsigned;
+
     /**
      * @param string $service Service name to use when signing
      * @param string $region  Region name to use when signing
+     * @param array $options  Array of configuration options used when signing
      */
-    public function __construct($service, $region)
+    public function __construct($service, $region, array $options = [])
     {
         $this->service = $service;
         $this->region = $region;
+        $this->unsigned = isset($options['unsigned']) ? $options['unsigned'] : false;
```

Comment:
Optional: `$options['unsigned']` should always be a boolean, so you could be more explicit in what values are accepted (e.g., `$this->unsigned = isset($options['unsigned']) && $options['unsigned'] === true`)